src/tests/ftest/container/container_check.py

```diff
@@ -0,0 +1,144 @@
+"""
+
+from command_utils import CommandFailure
+from daos_utils import DaosCommand
+from test_utils_pool import TestPool
+from test_utils_container import TestContainer
+from dfuse_utils import Dfuse
+from apricot import TestWithServers
+
+class DfuseContainerCheck(TestWithServers):
+    """Base Dfuse Container check test class.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DfuseContainerCheck object."""
+        super(DfuseContainerCheck, self).__init__(*args, **kwargs)
+        self.dfuse = None
+        self.pool = None
+        self.container = None
+
+    def setUp(self):
+        """Set up each test case."""
+        # Start the servers and agents
+        super(DfuseContainerCheck, self).setUp()
+
+    def tearDown(self):
+        """Tear down each test case."""
+        try:
+            if self.dfuse:
+                self.dfuse.stop()
+        finally:
+            # Stop the servers and agents
+            super(DfuseContainerCheck, self).tearDown()
+
+    def create_pool(self):
+        """Create a TestPool object to use with ior."""
+        # Get the pool params
+        self.pool = TestPool(
+            self.context, dmg_command=self.get_dmg_command())
+        self.pool.get_params(self)
+
+        # Create a pool
+        self.pool.create()
+
+    def start_dfuse(self):
+        """Create a DfuseCommand object to start dfuse.
+        """
+
+        # Get Dfuse params
+        self.dfuse = Dfuse(self.hostlist_clients, self.tmp)
+        self.dfuse.get_params(self)
+
+        # update dfuse params
+        self.dfuse.set_dfuse_params(self.pool)
+        self.dfuse.set_dfuse_cont_param(self.container)
+        self.dfuse.set_dfuse_exports(self.server_managers[0], self.client_log)
+
+        try:
+            # start dfuse
+            self.dfuse.run(False)
+        except CommandFailure as error:
+            self.log.error("Dfuse command %s failed on hosts %s",
+                           str(self.dfuse),
+                           self.dfuse.hosts,
+                           exc_info=error)
+            self.fail("Test was expected to pass but it failed.\n")
+
+    def test_dfusecontainercheck(self):
+        """Jira ID: DAOS-3635.
+
+        Test Description:
+            Purpose of this test is to try and mount different container types
+            to dfuse and check the behavior.
+        Use cases:
+            Create pool
+            Create container of type default
+            Try to mount to dfuse and check the behaviour.
+            Create container of type POSIX.
+            Try to mount to dfuse and check the behaviour.
+        :avocado: tags=all,small,full_regression,dfusecontainercheck
+        """
+        # get test params for cont and pool count
+        cont_types = self.params.get("cont_types", '/run/container/*')
+
+        # Create a pool and start dfuse.
+        self.create_pool()
```

Comment:
I believe you can use self.add_pool() here and you don't need the create_pool method.